cmd/run.go

```diff
@@ -120,6 +122,7 @@ func init() {
 	runCmd.Flags().StringSliceVarP(&dnsURIs, "dns-uri", "n", nil, "sets dns URI")
 	runCmd.Flags().StringVarP(&pacURI, "pac-uri", "p", "", "sets URI to PAC content, or directly, the PAC content")
 	runCmd.Flags().StringSliceVarP(&pacProxiesCredentials, "pac-proxies-credentials", "d", nil, "sets PAC proxies credentials using standard URI format")
+	runCmd.Flags().StringSliceVarP(&siteCredentials, "site-credentials", "d", nil, "sets site based credentials")
```

Comment:
typo here, `-d` is already used. I'd recommend using `StringSliceVar`, it's not a common option, no need to add a shortcut.

In general, I don't like passing credentials in this form: `host:port:user:pass`. I'd recommend using a more conventional `user:pass@url` (with the idea in mind that users of `forwarder` as a module would translate `host:port:user:pass` into `user:pass@url` 🤷‍♂️ . It's optional though, only if you agree.

Reply:
Oh yeah, that's a great idea. I was annoyed that I couldn't reuse the existing PAC auth code.